src/frontend/src/primitives/Text.tsx

```diff
@@ -0,0 +1,78 @@
+import type { HTMLAttributes } from 'react'
+import { RecipeVariantProps, cva, cx } from '@/styled-system/css'
+
+const text = cva({
+  base: {},
+  variants: {
+    variant: {
+      h1: {
+        textStyle: 'h1',
+        marginBottom: 'heading',
+      },
+      h2: {
+        textStyle: 'h2',
+        marginBottom: 'heading',
+      },
+      h3: {
+        textStyle: 'h3',
+        marginBottom: 'heading',
+      },
+      body: {
+        textStyle: 'body',
+      },
+      paragraph: {
+        textStyle: 'body',
+        marginBottom: 'paragraph',
+      },
+      small: {
+        textStyle: 'small',
+      },
+      inherits: {},
+    },
+    bold: {
+      true: {
+        fontWeight: 'bold',
+      },
+      false: {
+        fontWeight: 'normal',
+      },
+    },
+    italic: {
+      true: {
+        fontStyle: 'italic',
+      },
+    },
+    margin: {
+      false: {
+        margin: '0!',
+      },
+    },
+  },
+  defaultVariants: {
+    variant: 'inherits',
+  },
+})
+
+type TextHTMLProps = HTMLAttributes<HTMLElement>
+type TextElement =
+  | 'h1'
+  | 'h2'
+  | 'h3'
+  | 'h4'
+  | 'h5'
+  | 'h6'
```

Comment:
what does happen for those?

Reply:
We make a distincting between heading styles (where 1 2 3 are configured) and the Text HTML tag we might want to use. The `H` component is there to use headings styled as headings. The `Text` component lets you create any "text html tag" (the list you mention) with any "text style".